### DISCUSSION TITLE
I have a 2GB json file that times out after about 1000 records

### DISCUSSION THREAD

**Discussion by timcollins90** [halaxa/json-machine#104] [Q&A]
I'm parsing a 2GB json file into my website. The server is Ubuntu 22.04 with 4GB of Memory.  Shouldn't this server be able to parse through a 2GB json file without any problems? I keep getting timed out after about 1000 records have been parsed.

**Comment by halaxa** [comment]
It seems your memory leak is somewhere else. Try to only run the foreach loop without any code of yours in it. The other reason might be that one item in the JSON is just too big to fit into memory.
